ui/ui-guidelines.md

```diff
@@ -71,13 +100,133 @@ are reflected in how the codebase is organized.
   components (i.e. in `/views`), but for passing state via React Context that is
   application-wide, we use the `/context` folder
 
-### Tests
+## Build Process
+
+### All packages
+
+- Eslint enforces certain style and correctness rules, these can be ignored on a case
+  by case basis but usually are intended to help you avoid common JS pitfalls
+- Typescript is used to provide a stronger guarantee of correctness across the
+  application. Please avoid using escape hatches like `any`, prefer using `unknown`
+  and type narrowing instead.
+- Prettier is auto formatting all code to avoid bikeshedding on basic stylistic
+  concerns
+
+### Application
+
+- We use webpack to bundle our application, our config is heavily inspired by Create
+  React App, but ejected because we had the need to configure it to support specific
+  requirements for our project
+
+### Libraries
+
+- We use swc and tsc to build libraries.
+
+## Naming conventions
```

Comment:
Any other naming conventions worth calling out?

Reply:
This one really needs to be added here (https://github.com/perses/perses/blob/main/docs/dashboard.md) but it’s relevant for discussion on how to document it (more context in old #perses-dev thread with @Nexucis and came up in our weekly sync)

We’ve landed on snake_case for keys in our dashboard JSON / individual panel spec options, ex: allow_all_value, created_at, default_color, series_name_format, etc.

And PascalCase for the values in our dashboard JSON, ex: StatChart, LastNumber, Bottom, PercentDecimal, etc.

This trickles into our component API sometimes too, especially now that we’re talking about moving a lot of the mapping that happens in the panel, down into the presentational components, which is why I think it’s relevant here too

Reply:
I added a link to those docs as a starter for this PR. I'm inclined for us to update `dashboard.md` with some of those updates in a separate PR since this one is already getting pretty big and those changes aren't specific to the UI guidelines. Does that sound ok?